core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/BaseResourceSearchRequest.java

```diff
@@ -114,11 +114,11 @@ protected final Expression prepareQuery(RepositoryContext context) {
 		if (containsKey(OptionKey.SETTINGS)) {
 			final Collection<String> properties = getCollection(OptionKey.SETTINGS, String.class);
 			properties.forEach( property -> {
-				if (property.contains("#")) {
-					final String propertyName = property.split("\\#")[0];
+				if (property.contains(Resource.SETTINGS_DELIMITER)) {
+					final String propertyName = property.split(Resource.SETTINGS_DELIMITER)[0];
```

Comment:
The backslashes are now missing, but `#` doesn't seem to be a special regular expression character that would require escaping. The javadoc for this constant implies that it can be used for comments within an expression, although it might not be enabled by default: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/regex/Pattern.html#COMMENTS

Reply:
Exactly, that's why I was confident to drop the backslashes